apps/integrasjonstest/src/test/kotlin/no/nav/helsearbeidsgiver/inntektsmelding/integrasjonstest/utils/EndToEndTest.kt

```diff
@@ -271,72 +271,69 @@ abstract class EndToEndTest : ContainerTest() {
         forespoerselId: UUID,
         forespoerselSvar: ForespoerselFraBro?,
     ) {
-        var boomerang: JsonElement? = null
+        val messageSlot = slot<Map<Pri.Key, JsonElement>>()
 
         every {
             producer.send(
-                *varargAll { (key, value) ->
-                    if (key == Pri.Key.BOOMERANG) {
-                        boomerang = value
-                    }
-
-                    val erKorrektBehov = runCatching { value.fromJson(Pri.BehovType.serializer()) }.getOrNull() == Pri.BehovType.TRENGER_FORESPØRSEL
-
-                    (key == Pri.Key.BEHOV && erKorrektBehov) ||
-                        key in setOf(Pri.Key.FORESPOERSEL_ID, Pri.Key.BOOMERANG)
-                },
+                key = any(),
+                message = capture(messageSlot),
             )
         } answers {
-            publish(
-                Pri.Key.BEHOV to Pri.BehovType.TRENGER_FORESPØRSEL.toJson(Pri.BehovType.serializer()),
-                Pri.Key.LOESNING to
-                    ForespoerselSvar(
-                        forespoerselId = forespoerselId,
-                        resultat = forespoerselSvar,
-                        feil =
-                            if (forespoerselSvar == null) {
-                                ForespoerselSvar.Feil.FORESPOERSEL_IKKE_FUNNET
-                            } else {
-                                null
-                            },
-                        boomerang = boomerang.shouldNotBeNull(),
-                    ).toJson(ForespoerselSvar.serializer()),
-            )
-
-            boomerang = null
+            val message = messageSlot.captured
+            val behov = message[Pri.Key.BEHOV]?.runCatching { fromJson(Pri.BehovType.serializer()) }?.getOrNull()
+
+            if (behov == Pri.BehovType.TRENGER_FORESPØRSEL &&
+                !message.containsKey(Pri.Key.LOESNING) &&
```

Comment:
Er alle disse if sjekkene for å hindre loop av meldinger?

Reply:
Ja, men de forsikrer oss også om at det mockede svaret fra Storebror kun kommer ved rett spørsmål.